docs/madr/decisions/020-kds-redesign.md

```diff
@@ -0,0 +1,247 @@
+```
+
+How would a handshake work?
+
+1. The server exposes the RPC `handshake`.
+2. The client initiates a connection to the server.
+3. The client sends a request that includes the Kuma version and the supported KDS versions:
+```
+ kumaVersion: "2.1.0",
+ supportedKdsVersions: ["0.1.0", "0.2.0"],
+```
+4. Server checks the request and responds with the newest version working for both
+```
+ kdsVersion: "0.2.0"
+```
+5. The client needs to create RPC clients for the specific version.
+
+### Multitenant
+
+Global control-plane should support multitenant, that means we want to achieve that one global control-plane can handle many independent zone control-planes. One approach to accomplish this is by extracting tenant specifics from an authentication token. These details can then be transmitted along with a request to indicate the specific tenant and retrieve solely their deployment information. Another option is to supply this information via a header. It appears that accommodating multitenancy doesn't require additional work on the KDS side.
+
+### Create resources on the zone
+
+**That feature needs to be cover in a separate MADR**
+
+We intend to introduce a new feature that enables the creation of resources at the zone level. These resources will be exclusively applicable to the specific zone, and we will synchronize them to the global level as read-only. To distinguish the zone resources from global ones, we will prefix them with the name of the zone.
+
+```
+zone-1.example-timeout
+```
+
+### Resources
+Only policies that supports targetRef are going to be allowed.
+The following resources are the ones we should permit creation in the zone:
+* MeshAccessLog
+* MeshCircuitBreaker
+* MeshFaultInjection
+* MeshHealthCheck
+* MeshHttpRoute
+* MeshProxyPatch
+* MeshRateLimit
+* MeshRetry
+* MeshTimeout
+* MeshTrace
+* MeshTrafficPermission
+
+When there are two policies, one at the zone level and another at the global level, that impact the same dataplanes, the more specific policy takes precedence. In this scenario, the Zone policy would be the one utilized.
+
+Global policy:
+
+```yaml
+apiVersion: kuma.io/v1alpha1
+kind: MeshTimeout
+metadata:
+ name: timeout-global
+ namespace: kuma-system
+ labels:
+   kuma.io/mesh: default
+spec:
+ targetRef:
+   kind: Mesh
+ to:
+   - targetRef:
+       kind: Mesh
+     default:
+       idleTimeout: 20s
+       connectionTimeout: 2s
+```
+
+Zone policy:
+
+```yaml
+apiVersion: kuma.io/v1alpha1
+kind: MeshTimeout
+metadata:
+ name: timeout-zone
+ namespace: kuma-system
+ labels:
+   kuma.io/mesh: default
+spec:
+ targetRef:
+   kind: Mesh
+ to:
+   - targetRef:
+       kind: Mesh
+     default:
+       idleTimeout: 10s
+       connectionTimeout: 2s
+```
+
+The policy `timeout-zone` is going to be applied to all dataplanes in the default mesh in the zone. Also, worth adding that global control-plane added `kuma.io/zone: zone-name` to the resource when is synced.
```

Comment:
> Also, worth adding that global control-plane added `kuma.io/zone: zone-name` to the resource when is synced.

Should we remove this entire section given that it needs its own MADR? I come out of this section more confused and with more questions than never having read it.